azure-pipelines/build.yml

```diff
@@ -1,62 +1,94 @@
-  type: object
-  default:
-    vmImage: ubuntu-20.04
-- name: macOSPool
+# Whether this repo is localized.
+- name: EnableLocalization
+  type: boolean
+  default: false
+# Whether to run `dotnet format` as part of the build to ensure code style consistency.
+# This is just one of a a few mechanisms to enforce code style consistency.
+- name: EnableDotNetFormatCheck
+  type: boolean
+  default: true
+# This lists the names of the artifacts that will be published *from every OS build agent*.
+# Any new azure-pipelines/artifacts/*.ps1 script needs to be added to this list.
+# If an artifact is only generated or collected on one OS, it should NOT be listed here,
+# but should be manually added to the `outputs:` field in the appropriate OS job.
+- name: artifact_names
   type: object
   default:
-    vmImage: macOS-12
+    - build_logs
+    - coverageResults
+    - deployables
+    - projectAssetsJson
+    - symbols
+    - testResults
+    - test_symbols
+    - Variables
+# The Enable*Build parameters turn non-Windows agents on or off.
+# Their default value should be based on whether the build and tests are expected/required to pass on that platform.
+# Callers (e.g. Official.yml) *may* expose these parameters at queue-time in order to turn OFF optional agents.
 - name: EnableLinuxBuild
   type: boolean
   default: true
 - name: EnableMacOSBuild
   type: boolean
   default: true
-- name: EnableOptProf
-  type: boolean
-  default: false
-- name: IsOptProf
+
+##### 👆🏼 You MAY change the defaults above.
+##### 👇🏼 You should NOT change the defaults below.
+
+##### The following parameters are expected to be set by other YAML files that import this one.
+##### Those without defaults require explicit values to be provided by our importers.
+
+# Indicates whether the entrypoint file is 1ESPT compliant. Use this parameter to switch between publish tasks to fit 1ES or non-1ES needs.
+- name: Is1ESPT
   type: boolean
-  default: false
-- name: ShouldSkipOptimize
+
+- name: RealSign
   type: boolean
   default: false
+
 - name: RunTests
   type: boolean
   default: true
-- name: EnableLocalization
-  type: boolean
-  default: false
-- name: EnableCompliance
-  type: boolean
-  default: false
+
 - name: EnableAPIScan
   type: boolean
   default: false
-- name: EnableDotNetFormatCheck
+
+# This parameter exists to provide a workaround to get a build out even when no OptProf profiling outputs can be found.
+# Entrypoint yaml files like official.yml should expose this as a queue-time setting when EnableOptProf is true in this file.
+# The OptProf.yml entrypoint sets this parameter to true so that collecting profile data isn't blocked by a prior lack of profile data.
+- name: ShouldSkipOptimize
   type: boolean
-  default: true
-- name: artifact_names
+  default: false
+
+# The pool parameters are set to defaults that work in the azure-public AzDO account.
+# They must be overridden by callers for the devdiv AzDO account to use 1ES compliant pools.
```

Comment:
You could clarify if official.yml is already overriding as needed or if this is a request to change official.yml to use different pools